tests/core/InputParserTests.cs

```diff
@@ -0,0 +1,153 @@
+using System;
+using System.Linq;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Jupyter.Core
+{
+    internal class MockSymbolResolver : ISymbolResolver
+    {
+        private string[] magicSymbols = new[] { "%abc", "%def" };
+
+        public ISymbol Resolve(string symbolName)
+        {
+            if (this.magicSymbols.Contains(symbolName))
+            {
+                return new MagicSymbol
+                {
+                    Name = symbolName,
+                    Documentation = new Documentation(),
+                    Kind = SymbolKind.Magic,
+                    Execute = async (input, channel) => { return ExecutionResult.Aborted; }
```

Comment:
I think this should work as well?
```suggestion
                    Execute = async (input, channel) => ExecutionResult.Aborted;
```